WordPress/gutenberg
# theme.json schema: Allow object type on style properties

## What?

This PR adds support for objects, in addition to strings, for style properties in theme.json.

~Do we need to add support for objects on other properties, in addition to color?~ I've now added support for all style properties.

## Why?

In https://github.com/WordPress/gutenberg/pull/41696, we added the ability to reference values from other locations in the tree. However, currently, this isn't valid markup because the reference is an object rather than a string.

Fixes https://github.com/WordPress/gutenberg/issues/45889.

## How?

Add `object` to the supported types for style properties, and add a definition for `ref`, called `refComplete`.

## Testing Instructions




Add the following markup to a theme.json file:

```
		"elements": {
			"button": {
				"color": {
					"background": {
						"ref": "styles.color.text"
					},
					"text": {
						"ref": "styles.color.background"
					}
				}
			}
		}
```

Without this PR, you should see a warning in your text editor that the ref is an incorrect type (`Incorrect type. Expected "string".`). With this PR, this warning shouldn't show, at least for style values.

cc. @WordPress/block-themers
